src/coreclr/tools/aot/ILCompiler.MetadataTransform/Internal/Metadata/NativeFormat/Writer/NativeMetadataWriter.cs

```diff
@@ -964,61 +956,26 @@ public override string ToString()
     {
-        public static string FlagsToString<T>(this T value) where T : struct, Enum, IConvertible
-        {
-            var flags = Enum.GetValues<T>().Where(
-                eVal => (((IConvertible)eVal).ToInt32(null) != 0) && ((((IConvertible)value).ToInt32(null) & ((IConvertible)eVal).ToInt32(null)) == ((IConvertible)eVal).ToInt32(null)));
-            if (flags.Count() == 0)
-                return "";
-            else
-                return "[" + string.Join(" | ", flags.Select(Enum.GetName<T>)) + "] ";
-        }
-    }
-
-    public static class ListExtensions
-    {
-        public static T FirstOrDefault<T>(this List<T> list)
-        {
-            if (list.Count != 0)
-                return list[0];
-            return default(T);
-        }
-        public static T First<T>(this List<T> list) where T : class
+        public static string FlagsToString(this SignatureCallingConvention value)
         {
-            if (list.Count != 0)
-                return list[0];
-            return null;
-        }
-    }
-
-    public static partial class DictionaryExtensions
-    {
-        internal static T FirstOrDefault<T>(this Dictionary<string, T> dict)
-        {
-            if (dict.Count != 0)
-                foreach (var value in dict.Values)
-                    return value;
-            return default(T);
-        }
-        internal static T First<T>(this Dictionary<string, T> dict) where T : class
-        {
-            if (dict.Count != 0)
-                foreach (var value in dict.Values)
-                    return value;
-            return null;
-        }
-
-        internal static IEnumerable<T> AsSingleEnumerable<T>(this T value)
-        {
-            yield return value;
-        }
-    }
-
-    public static partial class SignatureHelpers
-    {
-        public static SZArraySignature AsSZArray(this MetadataRecord record)
-        {
-            return new SZArraySignature() { ElementType = record };
+            var values = new List<string>();
+            if ((value & SignatureCallingConvention.HasThis) == SignatureCallingConvention.HasThis)
+            {
+                value &= ~SignatureCallingConvention.HasThis;
+                values.Add(nameof(SignatureCallingConvention.HasThis));
+            }
+            if ((value & SignatureCallingConvention.ExplicitThis) == SignatureCallingConvention.ExplicitThis)
+            {
+                value &= ~SignatureCallingConvention.ExplicitThis;
+                values.Add(nameof(SignatureCallingConvention.ExplicitThis));
+            }
+            if ((value & SignatureCallingConvention.UnmanagedCallingConventionMask) != default)
+            {
+                value &= ~SignatureCallingConvention.UnmanagedCallingConventionMask;
+                values.Add((value & SignatureCallingConvention.UnmanagedCallingConventionMask).ToString());
+            }
+            values.Add(value.ToString());
+            return string.Join(" | ", values);
```

Comment:
Nit: This was wrapped in `[` and `]` before. Does the output look better without it?

Reply:
Well I totally missed that during refactor.